packages/react-components/react-swatch-picker-preview/src/components/SwatchPicker/SwatchPicker.types.ts

```diff
@@ -1,17 +1,58 @@
+import * as React from 'react';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { SwatchPickerContextValue } from '../../contexts/swatchPicker';
 
 export type SwatchPickerSlots = {
   root: Slot<'div'>;
 };
 
+export type SwatchPickerOnSelectionChangeEvent = React.MouseEvent | React.KeyboardEvent | React.ChangeEvent;
+
+export type SwatchPickerOnSelectionChangeData = {
+  selectedValue: string;
+};
+
+export type SwatchPickerSelectEventHandler = (
+  event: SwatchPickerOnSelectionChangeEvent,
+  data: SwatchPickerOnSelectionChangeData,
+) => void;
+
 /**
  * SwatchPicker Props
  */
-export type SwatchPickerProps = ComponentProps<SwatchPickerSlots> & {};
+export type SwatchPickerProps = ComponentProps<SwatchPickerSlots> & {
+  /**
+   * Default selected value
+   */
+  defaultSelectedValue?: string;
+
+  /**
+   * Triggers a callback when the value has been changed
+   */
+  // eslint-disable-next-line @nx/workspace-consistent-callback-type -- can't change type of existing callback
```

Comment:
Not relevant?

Reply:
why is this disabled ? any new components that don't implement old API should strictly adhere to our new event model type , am i missing something ?

Reply:
I'll rewrite it using new event model type